hibernate-core/src/main/java/org/hibernate/type/ProcedureParameterNamedBinder.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.type;
+
+import java.sql.CallableStatement;
+import java.sql.SQLException;
+
+import org.hibernate.HibernateException;
+import org.hibernate.engine.spi.SessionImplementor;
+
+/**
+ * Optional {@link Type} contract for implementations enabled
+ * to set store procedure OUT/INOUT parameters values by name.
+ *
+ * @author Andrea Boriero
+ */
+public interface ProcedureParameterNamedBinder extends Type {
```

Comment:
Thinking about this some more... maybe it is best to not have this extend Type explicitly.  The use-case I am thinking of is UserType.  It might be nice to allow UserType impls to also implement this contract.  We'd have to account for that in CustomType (the UserType wrapper adapting it to Type), but should be doable assuming we always rely on `ProcedureParameterNamedBinder#canDoSetting` rather than a simple `ProcedureParameterNamedBinder` type-check.  

If we do this, we should definitely do the same for `ProcedureParameterExtractionAware`

What do you think?